modules/transaction-payment/src/tests.rs

```diff
@@ -2176,34 +2210,41 @@ fn with_fee_call_validation_works() {
 
 			// with_fee_aggregated_path
 			let aggregated_path = vec![AggregatedSwapPath::Dex(vec![DOT, AUSD])];
-			assert_noop!(
+			assert_eq!(
```

Comment:
why this is changed?

Reply:
assert_noop  will throw a weird error: 
```
thread 'tests::with_fee_call_validation_works' panicked at modules/transaction-payment/src/tests.rs:2160:17:
assertion `left == right` failed: storage has been mutated
  left: [249, 149, 156, 205, 57, 147, 186, 208, 127, 112, 133, 173, 132, 93, 217, 62, 1, 173, 15, 252, 168, 86, 18, 7, 12, 133, 122, 249, 37, 36, 86, 78]
 right: [160, 125, 240, 154, 41, 126, 157, 145, 195, 135, 16, 84, 79, 200, 239, 206, 92, 159, 44, 8, 146, 97, 145, 12, 155, 52, 4, 30, 206, 186, 149, 211]
```

But use `assert_eq` shows it is `Err(TransactionValidityError::Invalid(InvalidTransaction::Payment))` actually.

Failed `ensure_can_charge_fee_with_call` may also set `OverrideChargeFeeMethod`, which may affects the judgment of `assert_noop`: https://github.com/paritytech/polkadot-sdk/blob/64d52f225e127575b51ab7fdf677783f5cd497f4/substrate/frame/support/src/lib.rs#L680-L693

I specific `ensure_can_charge_fee_with_call` to transactional, `assert_noop` works again.

Reply:
yeah the error means the storage is changed